src/ast/node.rs

```diff
@@ -1,6 +1,39 @@
 use serde::{Deserialize, Serialize};
+use serde_repr::{Deserialize_repr, Serialize_repr};
 
-pub type NodeID = u64;
+#[repr(i64)]
+#[derive(Clone, Copy, Debug, Deserialize_repr, Serialize_repr, PartialEq, Eq)]
+pub enum MagicVariableID {
+    Abi = -1,
+    Addmod = -2,
+    Assert = -3,
+    Block = -4,
+    Blockhash = -5,
+    Ecrecover = -6,
+    Gasleft = -7,
+    Keccak256 = -8,
+    Msg = -15,
+    Mulmod = -16,
+    Now = -17,
+    Require = -18,
+    Revert = -19,
+    Ripemd160 = -20,
+    Selfdestruct = -21,
+    Sha256 = -22,
+    Sha3 = -23,
+    Suicide = -24,
+    Super = -25,
+    Tx = -26,
+    Type = -27,
+    This = -28,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum NodeID {
```

Comment:
I think I'll go with `i64` to have parity with the Solidity's represenation (https://github.com/ethereum/solidity/blob/3f2cde9bd6073d27bab709d5759a4f55bcbeb006/libsolidity/ast/AST.h#L82)